drivers/interrupt_controller/intc_nxp_irqsteer.c

```diff
@@ -0,0 +1,522 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Driver for NXP's IRQ_STEER IP.
+ *
+ * Below you may find some useful information that will help you
+ * better understand how the driver works. The ">" sign is used
+ * to mark ideas that are considered important and should be taken
+ * note of.
+ *
+ * 1) What is the IRQ_STEER IP?
+ *	- in Zephyr terminology, the IRQ_STEER can be
+ *	considered an interrupt aggregator. As such, its
+ *	main goal is to multiplex multiple interrupt lines
+ *	into a single/multiple ones.
+ *
+ * 2) How does the IRQ_STEER IP work?
+ *	- below you may find a diagram meant to give you
```

Comment:
Does Zephyr has 100 chars per line limit? I think you can expand the lines up to that limit and the comments will be easier to read.

Reply:
Fixed, lmk if it's better.